### ISSUE TITLE
PF 2.0 dies with an error when trying to join threads

### ISSUE THREAD

**Issue by pbfrandsen** [brettc/partitionfinder#59]
This issue was brought up in the [PartitionFinder google group](https://groups.google.com/forum/#!topic/partitionfinder/L6BSZjXGiRc). Here is the error:

```
DEBUG    | 2015-08-10 15:08:36,659 | util       | Found and ignored Error when deleting file RAxML_log.a252a5828fceafbbba9684785ecbf27a_JTT+G.txt
DEBUG    | 2015-08-10 15:08:36,659 | util       | deleted 46 files
DEBUG    | 2015-08-10 15:08:36,690 | config     | Returning to original path: /scratch/kmeusema/PFinder
DEBUG    | 2015-08-10 15:08:36,690 | config     | Cleaning out all subsets (There are 1928)...
Traceback (most recent call last):
  File "/home/kmeusema/bin/partitionfinder-develop/PartitionFinderProtein.py", line 23, in <module>
    sys.exit(main.main("PartitionFinderProtein", "protein"))
  File "/home/kmeusema/bin/partitionfinder-develop/partfinder/main.py", line 370, in main
    run_analysis(cfg, options)
  File "/home/kmeusema/bin/partitionfinder-develop/partfinder/main.py", line 306, in run_analysis
    results = anal.analyse()
  File "/home/kmeusema/bin/partitionfinder-develop/partfinder/analysis.py", line 90, in analyse
    self.do_analysis()
  File "/home/kmeusema/bin/partitionfinder-develop/partfinder/analysis_method.py", line 276, in do_analysis
    start_result = self.analyse_scheme(start_scheme)
  File "/home/kmeusema/bin/partitionfinder-develop/partfinder/analysis.py", line 277, in analyse_scheme
    self.analyse_list_of_subsets(not_done)
  File "/home/kmeusema/bin/partitionfinder-develop/partfinder/analysis.py", line 255, in analyse_list_of_subsets
    self.run_threaded(tasks)
  File "/home/kmeusema/bin/partitionfinder-develop/partfinder/analysis.py", line 231, in run_threaded
    pool.join()
  File "/home/kmeusema/bin/partitionfinder-develop/partfinder/threadpool.py", line 100, in join
    raise self.exception
TypeError: __init__() takes exactly 3 arguments (1 given)
```

The full stdout is given as an attachment in that google group post.

**Comment by roblanf** [comment]
I'm fairly sure that this is related to https://github.com/brettc/partitionfinder/issues/60

I think what's happening is that errors are not propagating in a useful way when we use the threadpool. My bet is that when we fix https://github.com/brettc/partitionfinder/issues/60, this issue will be done.

**Comment by brettc** [comment]
Yep. The cause is parsing -- but the reported bug is because we don't
properly construct RaxmlError. On this now. Fix soon.

**Comment by brettc** [comment]
Fixed in 744bbedfb33c0ba7970d89cbdb7a94f40e1a4cf2
